From 56baabe054ebb1211cbb024794bc13b319bf4da9 Mon Sep 17 00:00:00 2001
From: "Yan, Xiangyang" <xiangyangx.yan@intel.com>
Date: Fri, 21 Apr 2017 01:13:53 +0000
Subject: [PATCH 3585/4301] trusty-ipc:tipc_msg_hdr structure: support large
 message transfer

commit 354c6029ea3981a43941016495583d83e113c79e from
git://git@github.com:01org/linux-intel-4.9.git

len field type of tipc_msg_hdr structure is u16 which will only
 handle message length of less than 64K. Change it to u32 to support
larger message.

Change-Id: I9f08d699842723224a10242d19165fa748a8c8b4
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-43927
Signed-off-by: Yan, Xiangyang <xiangyangx.yan@intel.com>
Reviewed-on: https://android.intel.com:443/575370
---
 drivers/trusty/trusty-ipc.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/trusty/trusty-ipc.c b/drivers/trusty/trusty-ipc.c
index d6765f1..9276ced 100644
--- a/drivers/trusty/trusty-ipc.c
+++ b/drivers/trusty/trusty-ipc.c
@@ -70,9 +70,9 @@ struct tipc_dev_config {
 struct tipc_msg_hdr {
 	u32 src;
 	u32 dst;
-	u32 reserved;
-	u16 len;
+	u32 len;
 	u16 flags;
+	u16 reserved;
 	u8 data[0];
 } __packed;
 
-- 
1.7.5.4

